### PR TITLE
add flag --extract tar file in podman cp

### LINKED ISSUE #2520: RFE Add --extract option to podman cp commands
**Is this a BUG REPORT or FEATURE REQUEST? (leave only one on its own line)**

kind feature

**Description**
This is a request for added functionality.  In #2517 we stopped the ability to extract a tar file during a `podman cp` command.  That's the right thing to do for a straight cp.  However I'd like to see a `podman cp --extract my.tar /mydir` option added such that if a tar file was being copied to the container, then it would be extracted there too.  Kind of a one stop shopping thing.

The old code around line 273 in that PR would have to be reinstated under an `if c.extract()` type of statement.

I'd suggest that if a destination argument was not given, the file would be extracted in the container's root directory, otherwise in the Destination directory given as an argument, making that directory first if it was not present. 

I've suggested `--extract` as the option name, but I'm not married to that.  IDK if Docker does something similar, if they do we may want to emulate that instead.

### PR DESCRIPTION
**Pull request by QiWang19**
fix #2520 
Use `--extract` to extract tar file in source argument to the destination directory
Signed-off-by: Qi Wang <qiwan@redhat.com>

### CODE REVIEW

File: docs/podman-cp.1.md [hunk 1]
```diff
@@ -54,6 +54,11 @@ You can also use : when specifying paths to a **SRC_PATH** or **DEST_PATH** on a
 If you use a : in a local machine path, you must be explicit with a relative or absolute path, for example:
 	`/path/to/file:name.txt` or `./file:name.txt`
 
+## OPTIONS
+
+**--extract**
+
+Extract the tar file into the destination directory. If the destination argument is not give, extract the tar file in root directory.
```
Comment (TomSweeneyRedHat) [inline docs/podman-cp.1.md]: suggest:

"Extract the tar file into the destination directory. If the destination option is not provided, extract the tar file into the root directory."

Comment (TomSweeneyRedHat) [reply docs/podman-cp.1.md]: oops, my bad, try 

"Extract the tar file into the destination directory. If the destination directory is not provided, extract the tar file into the root directory."

File: docs/podman-cp.1.md [hunk 2]
```diff
@@ -54,6 +54,11 @@ You can also use : when specifying paths to a **SRC_PATH** or **DEST_PATH** on a
 If you use a : in a local machine path, you must be explicit with a relative or absolute path, for example:
 	`/path/to/file:name.txt` or `./file:name.txt`
 
+## OPTIONS
```
Comment (TomSweeneyRedHat) [inline docs/podman-cp.1.md]: might be good to add an example in here too.

**Comment by giuseppe** [comment]
LGTM once you address the comments from @TomSweeneyRedHat

**Comment by QiWang19** [comment]
@giuseppe @TomSweeneyRedHat fixed :raising_hand_woman:

File: completions/bash/podman [hunk 3]
```diff
@@ -2308,6 +2308,15 @@ _podman_load() {
...
 }
 
+_podman_cp() {
+	local boolean_options="
+	  --help
+	  -h
+	  --extract
+     "
+	_complete_ "$boolean_options"
+}
```
Comment (TomSweeneyRedHat) [inline completions/bash/podman]: nice, didn't realize this was entirely missing....

**Comment by TomSweeneyRedHat** [comment]
LGTM

**Comment by mheon** [comment]
/approve
bot, retest this please

**Comment by jlebon** [comment]
bot, retest this please

**Comment by mheon** [comment]
bot, retest this please
/lgtm

**Comment by jlebon** [comment]
bot, retest this please

**Pull request merged**